rust_book_ex_8_3/src/string_ex.rs

```diff
@@ -0,0 +1,21 @@
+fn starts_with_vowel(s: &String) -> bool {
+    let vowels = vec!['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];
+
+    if let Some(first_char) = s.chars().next() {
+        if vowels.contains(&first_char) {
+            return true;
+        } else {
+            return false;
+        }
+    } else {
+        return false;
+    }
+}
+
+pub fn get_pig_latin(s: String) -> String {
+    if starts_with_vowel(&s) {
+        return format!("{}-hay", s);
+    } else {
+        return format!("{}-{}ay", s[1..].to_string(), s[0..1].to_string());
```

Comment:
Using `.to_string()` here is wrong, that's an extra allocation where you don't really need one.

You could just do

```rust
format!("{}-{}ay", &s[1..], &s[..1])
```